### ISSUE TITLE
Exception in error handling

### ISSUE THREAD

**Issue by zacwitte** [jkeyes/python-intercom#97]
```
Traceback (most recent call last): 
  File "/Users/zacwitte/Documents/workspace/handup/models/donor.py", line 279, in set_mixpanel
    icom_user = IntercomUser.create(**d)
  File "/usr/local/lib/python2.7/site-packages/intercom/api_operations/save.py", line 12, in create
    response = Intercom.post("/%s/" % (collection), **params)
  File "/usr/local/lib/python2.7/site-packages/intercom/__init__.py", line 171, in post
    return cls.request('POST', path, params)
  File "/usr/local/lib/python2.7/site-packages/intercom/__init__.py", line 163, in request
    method, cls.get_url(path), cls._auth, params)
  File "/usr/local/lib/python2.7/site-packages/intercom/request.py", line 40, in send_request_to_path
    return cls.parse_body(resp)
  File "/usr/local/lib/python2.7/site-packages/intercom/request.py", line 50, in parse_body
    cls.raise_application_errors_on_failure(body, resp.status_code)
  File "/usr/local/lib/python2.7/site-packages/intercom/request.py", line 102, in raise_application_errors_on_failure
    error_details, http_code)
  File "/usr/local/lib/python2.7/site-packages/intercom/request.py", line 113, in message_for_unexpected_error_with_type
    error_type = error_details['type']
KeyError: 'type'
```

**Comment by jkeyes** [comment]
Can you grab the response that Intercom is sending back? Can you let me know how I can reproduce the issue. Thanks.

**Comment by teeberg** [comment]
I sometimes get this error, but I can't seem to reproduce it. Here's the `decoded_body` of one failure:
```json
{
    "errors": [
        {
            "code": "unique_user_constraint",
            "message": "User already exists."
        }
    ],
    "request_id": "00000000-0000-0000-0000-000000000000",
    "type": "error.list"
}
```
I removed the request ID. Not sure what that's used for. :-) The command I run is:
```python
User.create(email=email, custom_attributes={'attribute': 'value'})
```
That's all the details I have so far. I'll report back when I'm able to find a scenario that reproduces it reliably.

**Comment by NickFranceschina** [comment]
seeing same thing here.   my response is 400:  

```
{"type":"error.list","request_id":null,"errors": [{"code":"parameter_not_found","message":"missing data parameter"}]}
```

blowing up same place as above with `KeyError`.... note how on this line you use the `error_details.get('type')` to access it in case it isn't there... 
 https://github.com/jkeyes/python-intercom/blob/master/intercom/request.py#L112

but then on this line you're using `error_details['type']` which causes the exception:
https://github.com/jkeyes/python-intercom/blob/master/intercom/request.py#L135

`parameter_not_found` isn't in the error code list

**Comment by jkeyes** [comment]
Thanks @teeberg and @NickFranceschina for raising this.

The error handling is supposed to look into the `error_details` alright. The problem is there is no `type` specified in there. Based on the Ruby lib (which was the basis for python-intercom), it should then fall back to the `code` in the error.

The two errors listed in this issue `unique_user_conflict` and `parameter_not_found` are not directly supported by the Ruby lib from what I can see.

I think the `unique_user_conflict` error should raise a `MultipleMatchingUsersError` and the `parameter_not_found` should raise a `BadRequestError`.

I'll get this fixed and I'll get in touch with Intercom so they can update the Ruby lib accordingly.

**Comment by jkeyes** [comment]
The fix will be in the 2.0.1 release.
